maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/util/DependencyScanner.java

```diff
@@ -111,19 +112,46 @@ private static void scanArtifact( File artifact, TestFilter<String, String> filt
         {
             for ( String groups : groupArtifactIds )
             {
+                // groupId:artifactId[:version[:type[:classifier]]]
```

Comment:
Maybe you could add `groupId:artifactId[:version[:type[:classifier]]]` to a regular JavaDoc on the top of Class or method and then this comment can be removed. The users will see it in public API JavaDoc for their better understanding.